server/src/main/java/org/opensearch/transport/InboundMessage.java

```diff
@@ -32,105 +32,71 @@
 
 package org.opensearch.transport;
 
-import org.opensearch.common.annotation.PublicApi;
 import org.opensearch.common.bytes.ReleasableBytesReference;
 import org.opensearch.common.lease.Releasable;
-import org.opensearch.common.lease.Releasables;
-import org.opensearch.common.util.io.IOUtils;
 import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.transport.nativeprotocol.NativeInboundMessage;
 
 import java.io.IOException;
 
 /**
  * Inbound data as a message
- *
+ * This api is deprecated, please use {@link org.opensearch.transport.nativeprotocol.NativeInboundMessage} instead.
  * @opensearch.api
  */
-@PublicApi(since = "1.0.0")
-public class InboundMessage implements Releasable {
+@Deprecated(since = "2.14.0")
+public class InboundMessage implements Releasable, ProtocolInboundMessage {
```

Comment:
This class is market as `@PublicApi(since = "1.0.0")` (OpenSearch annotation), since we deprecate it , we should replace it with`@DeprecatedApi(since = "2.14.0")` (OpenSearch annotation) 

```suggestion
@DeprecatedApi(since = "2.14.0")
```

Reply:
Done!